xl-v1.go

```diff
@@ -193,27 +199,37 @@ func (xl XL) DeleteVol(volume string) error {
 
 // ListVols - list volumes.
 func (xl XL) ListVols() (volsInfo []VolInfo, err error) {
-	// Pick the first node and list there always.
-	disk := xl.storageDisks[0]
-	volsInfo, err = disk.ListVols()
-	if err == nil {
-		return volsInfo, nil
+	quorumCount := 0
+	for _, disk := range xl.storageDisks {
+		volsInfo, err = disk.ListVols()
+		if err == nil {
+			quorumCount++
+		}
 	}
-	return nil, err
+	// Verify if we have enough quorum to list vols.
+	if quorumCount < xl.readQuorum {
+		return nil, errReadQuorum
+	}
+	return volsInfo, nil
```

Comment:
There is a chance of missing/added entries in the last `volsInfo` even after quorum is met.  As its not expected case, we would need to compare each returned `volsInfo` uniqueness along with quorum.